servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -528,7 +530,18 @@ public void handlerRemoved(ChannelHandlerContext ctx) {
 
         @Override
         public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
-            connection.nettyChannelPublisher.exceptionCaught(cause);
+            connection.nettyChannelPublisher.exceptionCaught(unwrapThrowable(cause));
+        }
+
+        /**
+         * Unwraps certain types of netty exceptions to directly expose its cause to improve debuggability.
+         */
+        private static Throwable unwrapThrowable(final Throwable t) {
+            final Throwable cause = t.getCause();
+            if (t instanceof DecoderException && cause instanceof SSLException) {
```

Comment:
nit: you could only do the actual unwrap if needed as `getCause()` may be expensive.

```java
final Throwable cause;

if (t instanceof DecoderException && ((cause = t.getCause()) instanceof SSLException) {
     return cause;
}
return t;
```

Reply:
Follow-up: https://github.com/apple/servicetalk/pull/1077